### PR TITLE
Replace deprecated references for analyzer 5.12

### PR DESCRIPTION
**Pull request by srawlins**

### CODE REVIEW

File: lib/src/rules/avoid_annotating_with_dynamic.dart [hunk 1]
```diff
@@ -81,7 +81,7 @@ class _Visitor extends SimpleAstVisitor<void> {
   }
 
   void _checkNode(NormalFormalParameter node, TypeAnnotation? type) {
-    if (type is NamedType && type.name.name == 'dynamic') {
+    if (type is NamedType && type.name2.lexeme == 'dynamic') {
```
Comment (scheglov) [inline lib/src/rules/avoid_annotating_with_dynamic.dart:84]: This is of course semantically the same as it was, but why now `type.type is DynamicType`?

File: lib/src/rules/use_to_and_as_if_applicable.dart [hunk 2]
```diff
@@ -55,7 +55,7 @@ bool _beginsWithAsOrTo(String name) {
 }
 
 bool _isVoid(TypeAnnotation? returnType) =>
-    returnType is NamedType && returnType.name.name == 'void';
+    returnType is NamedType && returnType.name2.lexeme == 'void';
```
Comment (scheglov) [inline lib/src/rules/use_to_and_as_if_applicable.dart:58]: Why not `returnType.type is VoidType`?

**Review by scheglov — APPROVED** [review]

File: lib/src/rules/avoid_annotating_with_dynamic.dart (hunk 1 above)
Comment (pq) [reply lib/src/rules/avoid_annotating_with_dynamic.dart:84]: Yes, please!

File: lib/src/rules/use_to_and_as_if_applicable.dart (hunk 2 above)
Comment (pq) [reply lib/src/rules/use_to_and_as_if_applicable.dart:58]: +1.

**Review by pq — APPROVED** [review]

File: lib/src/rules/avoid_types_on_closure_parameters.dart [hunk 3]
```diff
@@ -85,7 +85,7 @@ class AvoidTypesOnClosureParametersVisitor extends SimpleAstVisitor {
   @override
   void visitSimpleFormalParameter(SimpleFormalParameter node) {
     var type = node.type;
-    if (type is NamedType && type.name.name != 'dynamic') {
+    if (type is NamedType && type.name2.lexeme != 'dynamic') {
```
Comment (scheglov) [inline lib/src/rules/avoid_types_on_closure_parameters.dart:88]: Here too.

File: lib/src/rules/use_to_and_as_if_applicable.dart (hunk 2 above)
Comment (srawlins) [reply lib/src/rules/use_to_and_as_if_applicable.dart:58]: Done

File: lib/src/rules/avoid_annotating_with_dynamic.dart (hunk 1 above)
Comment (srawlins) [reply lib/src/rules/avoid_annotating_with_dynamic.dart:84]: Done

File: lib/src/rules/avoid_types_on_closure_parameters.dart (hunk 3 above)
Comment (srawlins) [reply lib/src/rules/avoid_types_on_closure_parameters.dart]: Done

**Pull request merged**